src/modules/deal-guarantee/dto/create-deal-guarantee-request.dto.ts

```diff
@@ -0,0 +1,79 @@
+import { PROPERTIES } from '@ukef/constants';
+import { ValidatedDateOnlyApiProperty } from '@ukef/decorators/validated-date-only-api-property.decorator';
+import { ValidatedNumberApiProperty } from '@ukef/decorators/validated-number-api-property.decorator';
+import { ValidatedStringApiProperty } from '@ukef/decorators/validated-string-api-property.decorator';
+import { DateOnlyString } from '@ukef/helpers/date-only-string.type';
+
+export type CreateDealGuaranteeRequest = CreateDealGuaranteeRequestItem[];
+
+export class CreateDealGuaranteeRequestItem {
+  @ValidatedStringApiProperty({
+    description: 'The identifier of the deal to create the guarantee for.',
+    example: '00000001',
+    minLength: 8,
+    maxLength: 8,
+  })
+  // TODO APIM-73: Should we remove dealIdentifier from the request body?
+  readonly dealIdentifier: string;
+
+  @ValidatedDateOnlyApiProperty({
+    description: `The date that this guarantee will take effect. This will be replaced by today's date if a date in the past is provided.`,
+  })
+  readonly effectiveDate: DateOnlyString;
+
+  @ValidatedStringApiProperty({
+    description: 'An ACBS party identifier.',
+    minLength: 8,
+    maxLength: 8,
+    example: '00000002',
+  })
+  readonly limitKey: string;
+
+  @ValidatedDateOnlyApiProperty({
+    description: 'The date that this guarantee will expire on.',
+  })
+  readonly guaranteeExpiryDate: DateOnlyString;
+
+  @ValidatedNumberApiProperty({
+    description: 'The maximum amount the guarantor will guarantee.',
+    minimum: 0,
+    maximum: 1e17, // TODO APIM-73: Is it okay that we lose precision at this magnitude?
```

Comment:
JS `number` doesn't have enough digits of precision to support numbers of this size to 2dp without losing precision

Reply:
Audrius and Abhi said on slack that we're happy to lose precision at this magnitude as the API will not be used with these values. We also want to remove the maximum validation as the existing legacy API does not have it